### PR TITLE
`SecretManager` changes

### PR DESCRIPTION
**Pull request by thibault-martinez**

### CODE REVIEW

File: client/src/error.rs [hunk 1]
```diff
@@ -181,6 +181,9 @@ pub enum Error {
     /// Input selection error.
     #[error("{0}")]
     InputSelection(#[from] InputSelectionError),
+    /// Missing chain to sign with.
+    #[error("missing chain to sign with")]
+    MissingChain,
```
Comment (Thoralf-M) [inline client/src/error.rs]: How about `MissingBip32Chain`?

Comment (thibault-martinez) [reply client/src/error.rs]: Yes sir

Comment (thibault-martinez) [reply client/src/error.rs]: Done

File: client/bindings/nodejs/test/client/offlineSigningExamples.spec.ts [hunk 2]
```diff
@@ -97,8 +97,8 @@ describe('Offline signing examples', () => {
 
         const unlock = await offlineClient.signatureUnlock(
             secretManager,
-            preparedTx.inputsData[0],
             hexToBytes(txEssenceHash),
+            preparedTx.inputsData[0].chain!,
```
Comment (thibault-martinez) [inline client/bindings/nodejs/test/client/offlineSigningExamples.spec.ts:101]: I really don't like that, we need to find another solution, maybe returning an error instead, idk

File: client/bindings/nodejs/types/bridge/client.ts [hunk 3]
```diff
@@ -166,10 +166,8 @@ export interface __SignatureUnlockMessage__ {
     name: 'signatureUnlock';
     data: {
         secretManager: SecretManager;
-        inputSigningData: IInputSigningData;
         transactionEssenceHash: Array<number>;
-        //TODO: Expose `RemainderData`.
-        remainderData: undefined;
+        chain: IBip32Chain;
```
Comment (kwek20) [inline client/bindings/nodejs/types/bridge/client.ts:170]: we need to expose chain?

**Review by kwek20 — DISMISSED** [review]

File: client/bindings/nodejs/test/client/offlineSigningExamples.spec.ts (hunk 2 above)
Comment (Thoralf-M) [reply client/bindings/nodejs/test/client/offlineSigningExamples.spec.ts:101]: But this is just a test

**Review by Thoralf-M — APPROVED** [review]

**Review by Thoralf-M — CHANGES_REQUESTED** [review]
Conflict

File: client/bindings/nodejs/test/client/offlineSigningExamples.spec.ts (hunk 2 above)
Comment (thibault-martinez) [reply client/bindings/nodejs/test/client/offlineSigningExamples.spec.ts:101]: Indeed

**Review by kwek20 — DISMISSED** [review]

**Review by Thoralf-M — DISMISSED** [review]

**Review by luca-moser — DISMISSED** [review]

**Review by Thoralf-M — APPROVED** [review]

**Pull request merged**